### DISCUSSION TITLE
How to enable Mutual TLS in minio for kubernetes?

### DISCUSSION THREAD

**Discussion by haylch** [minio/minio#19869] [Q&A]
Does adding custom CA, Server and Client certs in minio operator UI console is enough to enable the mutual TLS in minio running 
on kubernetes?
The one-way TLS works as the server did send certs for the client to verify in the TLS handshake process. But how do I enable mutual TLS? The connecting client does not seem to send certs for the server to verify. What steps am I missing here in order to turn on mutual TLS? 

My current settings in minio operator UI console
- TLS enabled
- AutoCert enabled
- Custom Certificates enabled with CA, Server and Client certs being added

My connecting client application has truststore and client cert keystore configured.

Any help is appreciated. Thanks.

**Comment by manju-rn** [comment]
Use Service mesh and let the mTLS managed by Service mesh - linkerd or Kuma
